lib/puma/dsl.rb

```diff
@@ -712,6 +712,17 @@ def on_booted(&block)
       @config.options[:events].on_booted(&block)
     end
 
+    # Code to run when Puma is stopped (works for both: single and clustered)
+    #
+    # @example
+    #   on_stopped do
+    #     puts 'Shutting down...'
+    #   end
+    def on_stopped(&block)
+      @options[:on_stopped] ||= []
+      @options[:on_stopped] << block
```

Comment:
This looks different to other events. And a simple test in config shoes it's being ignored completely. On the other hand, delegating this to events seem to work.

```suggestion
      @config.options[:events].on_stopped(&block)
```